packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -39,18 +38,36 @@ const isDebugEnabled = _debug('vite:deps').enabled
 const debounceMs = 100
 
 const depsOptimizerMap = new WeakMap<ResolvedConfig, DepsOptimizer>()
+const devSsrDepsOptimizerMap = new WeakMap<ResolvedConfig, DepsOptimizer>()
```

Comment:
Does this mean that SSR builds use `depsOptimizerMap`? If so it seems to be a bit confusing as we're only partially splitting SSR responsibilities in the optimizer. 

I'm fine for now so we can push out Vite stable, but something worth cleaning up later if true. (I still think a class is better for the deps optimizer that maybe I'll refactor someday 😄  )

Reply:
We could split the maps in: `depsOptimizerMap` and `ssrDepsOptimizerMap`, but it was more clear at the end to go with the design in the PR.
The important bit is that during build, we only have one DepsOptimizer (because you run build in ssr, or in client mode). During dev, we have two DepsOptimizer at the same time.
Open to PRs to clean up :)
I don't want to touch much a few days before the release.

Reply:
Thanks for the explanation! I think that make sense given the state of the code now, and we can definitely clean it up after stable. I think in the long run it'd be better for the deps optimizer in builds to have separate client/ssr variants since the output is different, but that's for another day.